File: leetcode1768.py
#solution 1
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
      ans: str = "";
      l1 = 0
      l2 = 0
      while l1 < len(word1) and l2 < len(word2):
        ans += word1[l1] + word2[l2]
        l1 += 1
        l2 += 1
      ans += word1[l1:] + word2[l2:]
      return ans
    
    #solution 2
    
    class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
      
      ans = "".join(list(map(lambda y:"".join(y),list(zip(word1,word2)))))
      
      ans += word1[len(ans)//2:] + word2[len(ans)//2:]
      return ans

File: leetcode1768.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
      int len1 = word1.length(),len2 = word2.length();
      int len = max(len1,len2);
      int i = 0;
      string ans;
      ans = ans + "";
      while(i < len){
        if(i < len1){
          ans += word1[i];
        }
        if(i < len2){
          ans += word2[i];
        }
        i ++;
      }
      return ans;
    }
};
